### PR TITLE
Fixing isNull() for link queries

### LINKED ISSUE #4856: Realm 3.4.0: exception for nested isNull() query
#### Goal
I had a nested _isNull_ query on a child object (like described in the [documentation](https://realm.io/docs/java/latest/#chaining-queries)). The query worked well until I updated to realm version 3.4.0.

#### Expected Results
I expected that the query is executed without exception using realm version 3.4.0 like it was with realm version 3.3.2.

Documentation of _isNull_:
```java
 /**
     * Tests if a field is {@code null}. Only works for nullable fields.
     * <p>
     * For link queries, if any part of the link path is {@code null} the whole path is considered to be {@code null}
     * e.g., {@code isNull("linkField.stringField")} will be considered to be {@code null} if either {@code linkField} or
     * {@code linkField.stringField} is {@code null}.
     *
     * @param fieldName the field name.
     * @return the query object.
     * @throws java.lang.IllegalArgumentException if the field is not nullable.
     * @see Required for further infomation.
     */
```

#### Actual Results
I received an exception using the nested _isNull_ query
```
java.lang.IllegalArgumentException: Illegal Argument: isNull() by nested query for link field is not supported.
 at io.realm.internal.TableQuery.nativeIsNull(Native Method)
 at io.realm.internal.TableQuery.isNull(TableQuery.java:611)
 at io.realm.RealmQuery.isNull(RealmQuery.java:205)
 at com.bz.realmchildquery.MainActivity.executeQuery(MainActivity.java:63)
```

#### Code Sample
You can download a sample project from: 
https://github.com/baloghz/RealmChildQuery.git

For demonstration I slightly modified the example from the [documentation](https://realm.io/docs/java/latest/#chaining-queries). A person can have one optional dog and the dog has an optional parameters (ate_at).

```java
public class Dog extends RealmObject {
    private Long ate_at;  // optional
    // getters & setters ...
}

public class Person extends RealmObject {
    private Dog dog; // optional
    // getters & setters ...
}
```
I would like to find the Persons whose dog's ate_at is null
```java
RealmQuery<Person> query = realm.where(Person.class).isNull("dog.ate_at");
```

Try to change realm versions and you can see that everything works fine in 3.3.2 and not in 3.4.0

#### Version of Realm and tooling
Realm version when the query worked: 3.3.2 
Realm version when the query gave an exception: 3.4.0

Realm sync feature enabled: no

Android Studio version: 2.3.3

Which Android version and device: API21, Samsung SM-N9005

### PR DESCRIPTION
**Pull request by kneth**
If the link query is on a nullable field, it seems to query the wrong field.

This is embarrassing bug, and the solution is simple: Follow links between tables and not queries links.

Fixes #4856

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 ### Bug Fixes
 
+* Fixed a bug related to link queries (#4856).
```
Comment (Zhuinden) [inline CHANGELOG.md]: well that's not very specific

**Comment by beeender** [comment]
https://github.com/realm/realm-java/issues/4864 could be fixed as well i think?

**Comment by kneth** [comment]
@beeender Maybe, it is probably worth adding a test inspired by #4864.

File: CHANGELOG.md (hunk 1 above)
Comment (kneth) [inline CHANGELOG.md]: Fixed a bug in `isNull()`, `isNotNull()`, `isEmpty()` and `isNotEmpty()` when queries involve nullable fields in link queries.

**Comment by Zhuinden** [comment]
Will there be a patch release with this change (3.4.1)?

**Comment by kneth** [comment]
@Zhuinden I guess so :smile:

**Pull request merged**